eventuals/task.h

```diff
@@ -412,14 +450,21 @@ struct _TaskFromToWith final {
                 std::move(fail),
                 std::move(stop));
             break;
-          case Action::Fail:
-            e->Fail(
-                interrupt,
-                std::move(exception.value()),
-                std::move(start),
-                std::move(fail),
-                std::move(stop));
+          case Action::Fail: {
+            // If 'Catches_' is not specified, that means that we
+            // will never invoke 'e->Fail', but we are using not constexpr
+            // Action as well, so compiler might try to go to 'e->Fail' path
+            // with wring error types, which might cause some errors.
```

Comment:
```suggestion
            // If 'Catches_' is empty then we will never dispatch
            // with an action of 'Action::Fail' but since 'action' is a runtime value
            // the compiler will assume that it's possible that we can call
            // 'e->Fail()', with what ever type we have for 'error' so to keep
            // the compiler from trying to compile that code path we need to
            // add the following 'if constexpr'.
```